src/lib/uri/uri.h

```diff
@@ -23,6 +32,10 @@ struct uri {
 	char *query;
 	char *fragment;
 	int host_hint;
+	/** Count of URI query parameters */
+	int param_count;
+	/** Different URI query parameters */
+	struct uri_query_param *params;
```

Comment:
You should also update `uri_format` in this patch to print query parameters and add a test. Please do.

Reply:
Never mind, it already adds query parameters to the output.

Reply:
Come to think of it, we do need to add `uri->query_params` instead of `uri->query`, because parameters can be added from a Lua table. So please update `uri_format`.

Reply:
I think again about it and i see such problem:
```lua
-- We have such URI:
u = uri.parse({"/tmp/1?xx=yy", params = {xx = "zz"}})
-- After parsing we have:
- host: unix/
  params:
    xx:
    - zz
  service: /tmp/1
  unix: /tmp/1
  query: xx=yy
-- After format now we have:
uri.format(u) -- unix/:/tmp/1?xx=yy
-- But should we have  ?? -- unix/:/tmp/1?xx=zz
...

```

Need to talk about it!